EasyCommands.Tests/ScriptTests/ScriptTest.cs

```diff
@@ -0,0 +1,182 @@
+﻿using IngameScript;
+using Malware.MDKUtilities;
+using Moq;
+using Sandbox.ModAPI.Ingame;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static IngameScript.Program;
+
+namespace EasyCommands.Tests.ScriptTests
+{
+    /// <summary>
+    /// Represents an instance of a ScriptTest. Constructing a new ScriptTest is akin to setting
+    /// up the test, as it instantiates required objects and performs basic setup.
+    /// 
+    /// This implements <see cref="IDisposable"/>, so any post-test cleanup is performed
+    /// upon destruction of the class. Please use this inside of a using statement.
+    /// </summary>
+    class ScriptTest : IDisposable
+    {
+        private GenericListDictionary mockedBlocksByName;
+        private List<IMyBlockGroup> mockedGroups;
+        private Mock<IMyGridTerminalSystem> gridMock;
+        private Program program;
+
+        public List<String> Logger { get; private set; }
+
+        public ScriptTest(String script)
+        {
+            mockedBlocksByName = new GenericListDictionary();
+            mockedGroups = new List<IMyBlockGroup>();
+            Logger = new List<String>();
+
+            // Init static fields that may have been modified from other tests, prepare for testing
+            Program.STATE = ProgramState.STOPPED;
+            Program.LOG_LEVEL = Program.LogLevel.SCRIPT_ONLY;
```

Comment:
nit: we might want to put these after the MDKFactor.CreateProgram call, lest we start moving some of this setup into the Program constructor

Reply:
Yeah, I moved 'em for now, then we can take another pass at changing these to instance variables.